src/lsm/manifest_log.zig

```diff
@@ -312,42 +348,78 @@ pub fn ManifestLogType(comptime Storage: type) type {
                 assert(table.address > 0);
 
                 if (label.event == .remove) {
-                    manifest.queue_for_compaction(block_address);
-
                     const table_removed =
                         manifest_log.tables_removed.fetchPutAssumeCapacity(table.address, {});
                     assert(table_removed == null);
                 } else {
                     if (manifest_log.tables_removed.get(table.address)) |_| {
-                        manifest.queue_for_compaction(block_address);
-
                         if (label.event == .insert) {
                             assert(manifest_log.tables_removed.remove(table.address));
                         }
                     } else {
                         var extent =
                             manifest_log.table_extents.getOrPutAssumeCapacity(table.address);
-                        if (extent.found_existing) {
-                            manifest.queue_for_compaction(block_address);
-                        } else {
+                        if (!extent.found_existing) {
                             extent.value_ptr.* = .{ .block = block_address, .entry = entry };
                             manifest_log.open_event(manifest_log, label.level, table);
                         }
                     }
                 }
             }
 
-            if (block_schema.entry_count < schema.Manifest.entry_count_max) {
-                manifest.queue_for_compaction(block_reference.address);
+            log.debug("{}: opened: checksum={} address={} entries={}", .{
+                manifest_log.superblock.replica_index.?,
+                block_checksum,
+                block_address,
+                block_schema.entry_count,
+            });
+
+            const checkpoint_state = &manifest_log.superblock.working.vsr_state.checkpoint;
+            if (checkpoint_state.manifest_head_address == block_address) {
+                // When we find the "head" block, stop iterating the linked list – any more blocks
```

Comment:
Wait, so the "head" block is the oldest one, and the "tail" block is the newest one? And `tail`' `previous` points (transitively) to the head? I was thinking the opposite this whole time! I guess it might make sense to duplicate comment from `ManifestReferences` onto `CheckPointState`, as the latter is the ground truth. 

I am also wondering if this is genuinelly flipped, mainly due to the `previous` link. If I start with the head, and follow previous links, I expect to reach the tail, and not vice verse.

The current naming _also_ makes sense to me from the perspective that we add entries to the `tail`. Maybe just call them `newest` and `oldest`?

Reply:
> If I start with the head, and follow previous links, I expect to reach the tail, and not vice verse.

Ahh that makes so much sense! (For context: I named the head/tail because I was thinking of it from the perspective of the manifest log's `log_block` ring buffers, where "log_block_{addresses,checksums}.tail" is the newest.)

I renamed them to `newest`/`oldest` instead -- that way they can't be mixed up.